modules/ROOT/pages/_partials/sync/server-auth.adoc

```diff
@@ -1,3 +1,142 @@
-= Authentication and Authorization
+= Authentication and Authorization using Keycloak
+
+Using a Keycloak service and the link:https://www.npmjs.com/package/@aerogear/voyager-keycloak[@aerogear/voyager-keycloak] module, it is possible to add security to a {sync-server} application. `@aerogear/voyager-keycloak` provides the following features out of the box.
+
+* Authentication - Ensure only authenticated users can access your server endpoints, including the main GraphQL endpoint.
+* Authorization - Use the `@hasRole()` directive within the GraphQL schema to implement role based access control (RBAC) on the GraphQL level.
+* Integration with GraphQL context - Use the `context` object within the GraphQL resolvers to access user credentials and several helper functions.
+
+= Prerequisites
+
+* There is a Keycloak service available.
+* You must add a valid `keycloak.json` config file to your project. 
+  ** Create a client for your application in the Keycloak administration console.
+  ** Click on the Installation tab.
+  ** Select *Keycloak OIDC JSON* for Format option, and click *Download*.
+
+= Setting Up Keycloak Protection in {sync-server}
+
+1. Import the Voyager Keycloak module
+
+[source,javascript]
+----
+const { KeycloakSecurityService } = require('@aerogear/voyager-keycloak')
+----
+
+2. Read the Keycloak config and pass it to initialise the `KeycloakSecurityService`.
+
+[source,javascript]
+----
+const keycloakConfig = JSON.parse(fs.readFileSync(path.resolve(__dirname, './path/to/keycloak.json')))
+const keycloakService = new KeycloakSecurityService(keycloakConfig)
+----
+
+3. Use the `keycloakService` instance to protect your app:
+
+[source,javascript]
+----
+const app = express()
+keycloakService.applyAuthMiddleware(app)
+----
+
+4. The last piece of work to do is telling Voyager server that the `keycloakService` above will be used as the security service:
+
+[source,javascript]
+----
+const voyagerConfig = {
+  securityService: keycloakService
+}
+const server = VoyagerServer(apolloConfig, voyagerConfig)
+----
+
+The link:https://github.com/aerogear/voyager-server/blob/master/examples/keycloak[Keycloak Example Server Guide] has an example server based off the instructions above and shows all of the steps needed to get it running.
+
+= Using the `@hasRole` Directive in Your Schema
```

Comment:
= Defining role based authorisation in your schema

or whatever... heading should reflect goal, not the action performed